ruff_lsp/server.py

```diff
@@ -1108,8 +1112,9 @@ async def format_document(params: DocumentFormattingParams) -> list[TextEdit] |
     # request itself can only act on a text document. A cell in a Notebook is
     # represented as a text document.
     document = Document.from_cell_or_text_uri(params.text_document.uri)
+
     result = await _run_format_on_document(document)
-    if result is None:
+    if result is None or result.exit_code != 0:
```

Comment:
I think we should also add this to `_lint_document_impl` for good measure.

Reply:
I think the patch in https://github.com/astral-sh/ruff-lsp/pull/314#issuecomment-1804875791 would apply to all uses

Reply:
Ahh you're right, we might exit 1 on lint errors.

Reply:
(In fact pretty sure we do.)